### PR TITLE
jsonnet: Drop cAdvisor metrics without (pod, namespace) label pairs.

### PR DESCRIPTION
**Pull request by PhilipGough**
This change intends to reduce cardinality by dropping cAdvisor metrics without (pod, namespace) label pairs.

EDIT: The initial implementation dropped excessive metrics and cost us the ability to effectively query system service resource usage ~Tested in minikube and the change drops 540 metrics. The list that is dropped can be seen here https://gist.github.com/PhilipGough/72c3a3f5f8b6676e6c09fc0dc9e9b033~

### CODE REVIEW

**Comment by paulfantom** [comment]
On a small cluster (6 nodes, ~90-100 containers) this reduced `count ({__name__=~"container.*"})` from 22776 to 10499 and I didn't notice any issues.

**Review by paulfantom — APPROVED** [review]
lgtm. One nit: let's move comment on why each metric is removed from git commit msg into code.

**Comment by tavin** [comment]
Anyone by chance have a link to an explanation of why some metrics don't have pod and namespace labels? I've run into this trying to analyze container_network_* metrics. Can't tell if it's a bug or some kind of "roll up" statistic.

**Comment by paulfantom** [comment]
The thing is that cAdvisor as such don't have a concept of `pod` or `namespace`, it is collecting data about every cgroup available in OS. It is kubernetes (or more precisely kubelet) who assigns relationships between `namespaces`, `pods`, and cgroups. However, since there are more cgroups on the host than there are pods, not all cgroups have that relationship set.

As an example, you can take any unit started by systemd. Since it is not governed by kubelet, it won't have `namespace` or `pod` association. However since it is started by systemd, it will get a cgroup and cAdvisor will report data about it.

TL;DR: there are more cgroups than pods and not all are related to kubernetes :)

**Comment by tavin** [comment]
thanks @paulfantom
